rs/xnet/payload_builder/src/impl_tests.rs

```diff
@@ -510,6 +510,96 @@ async fn validate_slice_above_msg_limit() {
     });
 }
 
+#[tokio::test]
+async fn validate_slice_above_signal_limit() {
+    with_test_replica_logger(|log| {
+        use ic_test_utilities::state_manager::encode_certified_stream_slice;
+
+        let state_manager = FakeStateManager::new();
+        let xnet_payload_builder = get_xnet_payload_builder_for_test(state_manager, log);
+        let validation_context = get_validation_context_for_test();
+
+        // `begin` and `end` of the reverse stream in this subnet.
+        const REVERSE_STREAM_BEGIN: u64 = 13;
+        const REVERSE_STREAM_END: u64 = REVERSE_STREAM_BEGIN + 10;
+
+        // `begin`, `end` and `signals_end` such that the stream on the remote subnet has maximum
+        // size and nothing is gc'ed on this subnet.
+        const STREAM_BEGIN: u64 = 20;
+        const MAX_STREAM_END: u64 = STREAM_BEGIN + MAX_STREAM_MESSAGES as u64;
+        const SIGNALS_END: u64 = REVERSE_STREAM_BEGIN;
+        // `begin` of `messages` in the stream slice.
+        const MESSAGE_BEGIN: u64 = STREAM_BEGIN + 30;
+
+        // State of an `Application` subnet with a stream for `SUBNET_1`.
+        let mut state = ReplicatedState::new(OWN_SUBNET_ID, SubnetType::Application);
+        state.with_streams(btreemap! {
+            SUBNET_1 => generate_stream(&StreamConfig {
+                message_begin: REVERSE_STREAM_BEGIN,
+                message_end: REVERSE_STREAM_END,
+                signal_end: MESSAGE_BEGIN,
+            }),
+        });
+
+        // An oversized stream to take slices from.
+        let stream = generate_stream(&StreamConfig {
+            message_begin: STREAM_BEGIN,
+            message_end: MAX_STREAM_END + 10,
+            signal_end: SIGNALS_END,
+        });
+
+        // Helper for validating a generated slice from `SUBNET_1` taken from `stream`
+        // with messages starting end ending at the given indices.
+        let validate_slice = |slice_begin: u64, slice_end: u64, state| {
+            let slice = stream.slice(slice_begin.into(), Some((slice_end - slice_begin) as usize));
```

Comment:
If the stream is oversized (but not the slice), do we generate a warning anywhere?